metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/utils/EntityDao.java

```diff
@@ -0,0 +1,263 @@
+
+  //Table names
+  private static final String AGENT_TABLE = "people";
+  private static final String CONCEPT_TABLE = "concept";
+  private static final String PLACE_TABLE = "place";
+  private static final String TIMESPAN_TABLE = "period";
+  public static final String ORGANIZATION_TABLE = "organization";
+  private final MongoClient mongoClient;
+
+
+  public EntityDao(MongoClient mongoClient, String databaseName) {
+    this.mongoClient = mongoClient;
+
+    // Register the mappings and set up the data store.
+    final Morphia morphia = new Morphia();
+    morphia.map(MongoTermList.class);
+    morphia.map(AgentTermList.class);
+    morphia.map(ConceptTermList.class);
+    morphia.map(PlaceTermList.class);
+    morphia.map(TimespanTermList.class);
+    morphia.map(OrganizationTermList.class);
+    this.datastore = (AdvancedDatastore) morphia.createDatastore(this.mongoClient, databaseName);
+  }
+
+  public <T extends MongoTermList<S>, S extends AbstractEdmEntityImpl> MongoTermList<S> findTermListByField(
+      Class<T> mongoTermListType, String fieldName, String fieldValue) {
+    return ExternalRequestUtil.retryableExternalRequestConnectionReset(
+            () -> this.datastore.find(mongoTermListType).filter(fieldName, fieldValue).first());
+  }
+
+
+  private MongoTerm findMongoTermByField(String entityType, String fieldName, String fieldValue) {
+    return ExternalRequestUtil.retryableExternalRequestConnectionReset(
+        () -> this.datastore.find(entityType, MongoTerm.class).filter(fieldName, fieldValue)
+            .first());
+  }
+
+  public List<MongoTerm> getAllMongoTerms(EntityType entityType) {
+    Query<MongoTerm> query = this.datastore.createQuery(getTableName(entityType), MongoTerm.class);
+    return getListOfQuery(query, null);
+  }
+
+  public Date getDateOfLastModifiedEntity(EntityType entityType) {
+    Query<MongoTermList> query = datastore.createQuery(MongoTermList.class);
+    query.filter(ENTITY_TYPE_FIELD, getEntityType(entityType));
+    query.order(Sort.descending(MODIFIED_FIELD));
+    final MongoTermList mongoTermList = ExternalRequestUtil
+        .retryableExternalRequestConnectionReset(query::first);
+
+    Date dateModified = null;
+    if (Objects.nonNull(mongoTermList)) {
+      dateModified = mongoTermList.getModified();
+    }
+    return dateModified;
+  }
+
+  public String saveTermList(MongoTermList<? extends AbstractEdmEntityImpl> mongoTermList) {
+    Key<MongoTermList<? extends AbstractEdmEntityImpl>> datasetKey = ExternalRequestUtil
+        .retryableExternalRequestConnectionReset(
+            () -> this.datastore.save(mongoTermList));
+    return datasetKey == null ? StringUtils.EMPTY : datasetKey.getId().toString();
+  }
+
+  public void saveMongoTerms(List<MongoTerm> mongoTerms, EntityType entityType) {
+    ExternalRequestUtil.retryableExternalRequestConnectionReset(
+        () -> this.datastore.save(getTableName(entityType), mongoTerms));
```

Comment:
save is deprecated.

Reply:
Yes there is no other way at the moment, we'll need to change the db structure to fix this